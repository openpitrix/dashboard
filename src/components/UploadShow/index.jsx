import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { observer } from 'mobx-react';
import classnames from 'classnames';
import { translate } from 'react-i18next';

import { Icon } from 'components/Base';

import styles from './index.scss';

@translate()
@observer
export default class UploadShow extends Component {
  static propTypes = {
    className: PropTypes.string,
    errorMessage: PropTypes.string,
    fileName: PropTypes.string,
    isLoading: PropTypes.bool,
    uploadStatus: PropTypes.string
  };

  render() {
    const {
      className,
      isLoading,
      errorMessage,
      uploadStatus,
      fileName,
      t
    } = this.props;

    if (isLoading) {
      return (
        <div className={classnames(styles.uploading, className)}>
          <Icon name="loading" size={48} type="dark" className={styles.icon} />
          <p className={styles.note}>{t('file_format_loading')}</p>
        </div>
      );
    }

    return (
      <div
        className={classnames(className, styles.uploadShow, {
          [styles.bodyBg]: !!errorMessage
        })}
      >
        {uploadStatus !== 'ok'
          && !errorMessage && (
            <div>
              <Icon
                name="upload"
                size={48}
                type="dark"
                className={styles.icon}
              />
              <p className={styles.note}>{t('file_format_note')}</p>
            </div>
        )}

        {uploadStatus !== 'ok'
          && errorMessage && (
            <div className={styles.uploadError}>
              <Icon name="error" size={48} className={styles.icon} />
              <p>
                {errorMessage}
                <span className={styles.errorLink}>
                  「{t('Upload again')}」
                </span>
              </p>
            </div>
        )}
        {uploadStatus === 'ok'
          && !errorMessage && (
            <div className={styles.uploadSuccess}>
              <Icon name="checked-circle" size={48} className={styles.icon} />
              <p className={styles.successText}>
                {t('File')}
                <span className={styles.uploadFileName}>{fileName}</span>
                {t('Successful upload')}
              </p>
            </div>
        )}
      </div>
    );
  }
}
