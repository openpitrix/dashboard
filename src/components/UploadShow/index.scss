@import '~scss/vars';

@mixin disable-hover($opacity) {
  opacity: $opacity;
  cursor: default;
  &:hover {
    border-color: transparent;
    .intro {
      display: block;
    }
    a {
      display: none;
    }
  }
}

.uploadShow {
  position: relative;
  display: inline-block;
  box-sizing: border-box;
  width: 100%;
  padding: 32px 24px;
  border-radius: 2px;
  background-color: $N0;
  border: dashed 1px $N30;
  text-align: center;
  cursor: pointer;

  &.bodyBg {
    background-color: $background-color;
    border: solid 1px $background-color;
  }

  .note {
    text-align: center;
    font-size: 12px;
    line-height: 20px;
    color: $N100;
  }

  .icon {
    margin-bottom: 8px;
  }
}

.uploading {
  width: 100%;
  padding: 36px 50px;
  background-color: $N0;
  border-radius: 2px;
  box-shadow: 0 1px 4px 0 rgba(73, 33, 173, 0.06),
    0 4px 8px 0 rgba(35, 35, 36, 0.04);
  text-align: center;
  cursor: not-allowed;

  .icon {
    margin-bottom: 8px;
  }

  .note {
    text-align: center;
    font-size: 12px;
    line-height: 20px;
    color: $N100;
  }
}

.uploadError {
  font-size: 12px;
  line-height: 20px;
  color: $N75;
  text-align: center;

  .icon svg {
    --primary-color: #{$R400};
    --secondary-color: #{$R30};
  }

  .errorLink {
    color: $P75;
  }
}

.uploadSuccess {
  text-align: center;

  .icon svg {
    --primary-color: #{$G100};
    --secondary-color: #{$G30};
  }

  .successText {
    color: $N75;
    .uploadFileName {
      color: $N500;
      padding: 0 6px;
    }
  }
}
