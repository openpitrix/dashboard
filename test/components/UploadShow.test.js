import UploadShow from 'components/UploadShow';

describe('Component UploadShow', () => {
  it('basic render', () => {
    const wrapper = render(<UploadShow />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Upload error', () => {
    const wrapper = render(<UploadShow errorMessage="Upload error" />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Upload Ok', () => {
    const wrapper = render(<UploadShow uploadStatus="ok" />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Upload isLoading', () => {
    const wrapper = render(<UploadShow isLoading />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
