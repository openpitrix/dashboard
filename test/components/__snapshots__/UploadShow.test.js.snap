// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component UploadShow Upload Ok 1`] = `
<div
  class="uploadShow"
>
  <div
    class="uploadSuccess"
  >
    <span
      class="icon icon-checked-circle icon"
      style="width:48px;height:48px"
    >
      <svg
        class="qicon qicon-checked-circle qicon-light"
      >
        <use
          href="#qui-checked-circle"
        />
      </svg>
    </span>
    <p
      class="successText"
    >
      File
      <span
        class="uploadFileName"
      />
      Successful upload
    </p>
  </div>
</div>
`;

exports[`Component UploadShow Upload error 1`] = `
<div
  class="uploadShow bodyBg"
>
  <div
    class="uploadError"
  >
    <span
      class="icon icon-error icon"
      style="width:48px;height:48px"
    >
      <svg
        class="qicon qicon-error qicon-light"
      >
        <use
          href="#qui-error"
        />
      </svg>
    </span>
    <p>
      Upload error
      <span
        class="errorLink"
      >
        「Upload again」
      </span>
    </p>
  </div>
</div>
`;

exports[`Component UploadShow Upload isLoading 1`] = `
<div
  class="uploading"
>
  <span
    class="icon icon-loading icon"
    style="width:48px;height:48px"
  >
    <svg
      class="qicon qicon-loading qicon-dark"
    >
      <use
        href="#qui-loading"
      />
    </svg>
  </span>
  <p
    class="note"
  >
    file_format_loading
  </p>
</div>
`;

exports[`Component UploadShow basic render 1`] = `
<div
  class="uploadShow"
>
  <div>
    <span
      class="icon icon-upload icon"
      style="width:48px;height:48px"
    >
      <svg
        class="qicon qicon-upload qicon-dark"
      >
        <use
          href="#qui-upload"
        />
      </svg>
    </span>
    <p
      class="note"
    >
      file_format_note
    </p>
  </div>
</div>
`;
